### ISSUE TITLE
CouplingBetweenObjects excluding some pattern

### ISSUE THREAD

**Issue by dakorpar** [phpmd/phpmd#637]
We often have situation that coupling between objects raises because class is using multiple enum classes and that doesn't really increase complexity. Also another case is some static util classes, for instance if we use Strings::replace() instead of str_replace, so it would be really benefitial if we could somehow put classes on ignore list

**Comment by ravage84** [comment]
This could be just as StaticAccess's exceptions list:

https://phpmd.org/rules/cleancode.html#staticaccess

@dakorpar would you be willing to put up a PR for this? We can help yoyu with that.

**Comment by dakorpar** [comment]
I could try, but from what I saw rule only reads cbo metric, so not really sure where to implement this and how

**Comment by kylekatarnls** [comment]
Good point, so the options are: rewrite the rule using some `collect*` (manually count the `use` at the top of the file). Or edit PDepend to have some ignore filtering in CBO. I guess the first option is more relevant.

**Comment by dakorpar** [comment]
I believe just counting use is not enough, this needs to check all code... sometimes there can be FQN in code and then this won't work
